src/update-changelog.ts

```diff
@@ -264,28 +289,31 @@ export async function updateChangelog({
       changelog.addRelease({ version: currentVersion });
     }
 
-    if (hasUnreleasedChanges) {
+    const hasUnreleasedChangesToRelease =
+      getKnownPropertyNames(changelog.getUnreleasedChanges()).length > 0;
+    if (hasUnreleasedChangesToRelease) {
       changelog.migrateUnreleasedChangesToRelease(currentVersion);
     }
+  }
 
-    const newChangeEntries = newCommits.map(({ prNumber, description }) => {
-      if (prNumber) {
-        const suffix = `([#${prNumber}](${repoUrl}/pull/${prNumber}))`;
-        return `${description} ${suffix}`;
-      }
-      return description;
-    });
+  const newChangeEntries = await getNewChangeEntries({
+    mostRecentTag,
+    repoUrl,
+    loggedPrNumbers: getAllLoggedPrNumbers(changelog),
+    projectRootDirectory,
+  });
 
-    for (const description of newChangeEntries.reverse()) {
-      changelog.addChange({
-        version: isReleaseCandidate ? currentVersion : undefined,
-        category: ChangeCategory.Uncategorized,
-        description,
-      });
-    }
```

Comment:
This is the bracket that caused the bug. Moving it to above the `getNewChangeEntries` call resolves the error.